physics/noahmpdrv.F90

```diff
@@ -358,6 +400,47 @@ subroutine noahmpdrv_run                                       &
   character(len=*)    ,                    intent(out)   :: errmsg
   integer             ,                    intent(out)   :: errflg
 
+  real(kind=kind_phys), dimension(:)  , intent(out), optional :: canopy_heat_storage_ccpp ! within-canopy heat [W/m2]
```

Comment:
I'm confused by these optional outputs. Can you describe how they will be used? They all have a `_ccpp` suffix, yet none of them can actually be used in the CCPP because there is no metadata and they are not allocated in FV3.

Reply:
We are using noahmpdrv in other applications, e.g., the noahmp component model and land only driver. Basically our attempt to do hierarchical land only simulations using the noahmpdrv interface to noahmplsm so that we can test the noahmpdrv code as well. In those applications we want access to more output of noahmp, many of them sub-tile fluxes and states. We would like the flexibility to not allocate all of these in FV3.

I'll admit the _ccpp suffix is not the best (it seems so in my mind at some point). We can change them to something like _vector if you want. After this PR, we will make an effort to create a submodule for noahmp. During this process we will rewrite and clean up noahmpdrv, e.g., more cleanly associating the scalars coming out of noahmp to vectors of noahmpdrv, so we can write associations like
`temperature_ground_vector(i) = temperature_ground`
instead of
`tgxy(i) = temperature_ground`

Reply:
OK, thanks. No need to change if it makes sense to you -- just trying to understand what was done as part of the review process.